File: src/main/java/seremis/geninfusion/soul/gene/newAI/ownerHurtByTarget/GeneAIOwnerHurtByTarget.java
package seremis.geninfusion.soul.gene.newAI.ownerHurtByTarget;

import seremis.geninfusion.api.soul.IAllele;
import seremis.geninfusion.api.soul.lib.Genes;
import seremis.geninfusion.soul.MasterGene;
import seremis.geninfusion.soul.allele.AlleleBoolean;

public class GeneAIOwnerHurtByTarget extends MasterGene {

    public GeneAIOwnerHurtByTarget() {
        addControlledGene(Genes.GENE_AI_OWNER_HURT_BY_TARGET_INDEX);
    }

    @Override
    public Class<? extends IAllele> possibleAlleles() {
        return AlleleBoolean.class;
    }
}

File: src/main/java/seremis/geninfusion/soul/gene/newAI/GeneAIOwnerHurtByTarget.scala
package seremis.geninfusion.soul.gene.newAI

import seremis.geninfusion.api.soul.EnumAlleleType
import seremis.geninfusion.api.soul.lib.Genes
import seremis.geninfusion.soul.MasterGene

class GeneAIOwnerHurtByTarget extends MasterGene(EnumAlleleType.BOOLEAN) {
    addControlledGene(Genes.GENE_AI_OWNER_HURT_BY_TARGET_INDEX)
}